pkg/core/pipeline/pullRequest/main.go

```diff
@@ -0,0 +1,134 @@
+package pullRequest
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/sirupsen/logrus"
+	"github.com/updatecli/updatecli/pkg/core/pipeline/scm"
+	"github.com/updatecli/updatecli/pkg/plugins/github"
+)
+
+var (
+	// ErrWrongConfig is returned when a pullrequest has missing mandatory attributes.
+	ErrWrongConfig = errors.New("wrong pull request configuration")
+)
+
+// PullRequestHandler interface defines required functions to be an pullRequest
+type PullRequestHandler interface {
+	CreatePullRequest(title, changelog, pipelineReport string) error
+}
+
+// Config define pullRequest provided via an updatecli configuration
+type Config struct {
+	Title   string      // Define the pullRequest Title
+	Kind    string      // Define the pullRequest kind
+	Spec    interface{} // Define specific parameters
+	ScmID   string      // Reference a scm configuration
```

Comment:
```suggestion
	ScmID   string      `yaml:"scmID"` // References a scm configuration
```
In order to stay coherent with the other classes. Please not that it might need to update the symbol on other locations

Reply:
I wouldn't enforce scmID as it defines a strong mapping between a updatecli configuration and the Golang struct.
Whithout "`yaml:"scmID"`", we could use `scmid` in the updatecli configuration. We aren't good yet at detecting configuration errors.

Reply:
It seems that you are right and I was wrong